addons/website_sale/tests/test_website_sale_image.py

```diff
@@ -229,3 +229,21 @@ def test_01_admin_shop_zoom_tour(self):
         self.assertEqual(images[1].image_1920, red_image)
         self.assertEqual(images[2].image_1920, image_gif)
         self.assertEqual(images[3].image_1920, image_svg)
+
+        # CASE: When uploading a product variant image, we don't want the default_product_tmpl_id from the context to be applied
+        additionnal_context = {'default_product_tmpl_id': template.id}
+
+        product = self.env['product.product'].create({
+            'product_tmpl_id': template.id,
+            'image_1920': green_image,
+        })
+
+        self.env['product.image'].with_context(**additionnal_context).create([{
+            'name': 'Test Image',
+            'image_1920': red_image,
+            'product_variant_id': product.id,
+        }])
+
+        product_image = self.env['product.image'].search([], limit=1, order='id desc')
+        self.assertEqual(product_image.product_tmpl_id.id, False)
+        self.assertEqual(product_image.product_variant_id.id, product.id)
```

Comment:
You can compare recordsets directly, without having to type `.id`. It compares their ids but also does an additional type check. It does not really matter in this case, just for information.

Reply:
Thanks for the tip!